### DISCUSSION TITLE
why is not possible to config all options used for db connection in supper

### DISCUSSION THREAD

**Discussion by n4an** [prisma/prisma#10325] [Q&A]
```
constructor(private configService: ConfigService) {
    super({
    datasources: {
        db: {
          url: configService.get<string>('app.database.datasources.db.url')
        }
      },
})

```

why is not possible to config all options in supper 

```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
}
```

**Comment by janpio** [comment]
What is `super`? 
What options are you missing to configure?

**Reply by n4an** [reply]
@janpio  can't set provider 

```
@Injectable()
export class DatabaseService extends PrismaClient implements OnModuleInit, OnModuleDestroy {
  private readonly logger: Logger

  constructor(private configService: ConfigService) {
    super({
      log: [
        { emit: 'event', level: 'query' },
        { emit: 'event', level: 'info' },
        { emit: 'stdout', level: 'warn' },
        { emit: 'stdout', level: 'error' }
      ],
      errorFormat: 'colorless',
      datasources: {
        db: {
          provider: 'mysql', // <============== here
          url: configService.get<string>('app.database.datasources.db.url')
        }
      },
      rejectOnNotFound: true
    })

    this.logger = new Logger(DatabaseService.name)
  }
```

**Reply by janpio** [reply]
Ah. The `provider` value is required on build time of the Prisma Client  (when you run `prisma generate`) , as that already decides which functionality is included and which is not. The `url` on the other hand, can be configured on runtime (when you run and execute your app) as that is just dynamic string that influences where the Client connects to - but nothing about the Client itself.

**Reply by n4an** [reply]
but i can't use `configService.get<string>('app.database.datasources.db.provider')`. and in config i can set mysql or postgress. i can do that switch at any time

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
